lib/board.rb

```diff
@@ -0,0 +1,63 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def input_to_index(input)
+    input.to_i - 1
+  end
+
+  def position(input)
+    input = input_to_index(input)
+    cells[input]
+  end
+
+  def full?
+    cells.all? do |cell|
+      cell == "X" || cell == "O"
+    end
+  end
+
+  def turn_count
+    turns = 0
+    cells.each do |cell|
+      if cell == "X" || cell == "O"
+        turns += 1
+      end
+    end
+    turns
+  end
+
+  def taken?(input)
+    if position(input) == "X" || position(input) == "O"
+      true
+    else
+      false
+    end
+  end
+
+  def valid_move?(input)
+    if !taken?(input) && input.to_i.between?(1,9)
```

Comment:
No need for an if statement. `!taken?(input) && input.to_i.between?(1,9)` wil return a boolean by itself.